src/Control/Email/Email.php

```diff
@@ -316,430 +221,142 @@ private function getDefaultFrom(): string
-    {
-        $this->getSwiftMessage()->setSubject($subject);
-
-        return $this;
-    }
-
-    /**
-     * @return int
-     */
-    public function getPriority()
-    {
-        return $this->getSwiftMessage()->getPriority();
-    }
-
-    /**
-     * @param int $priority
-     * @return $this
-     */
-    public function setPriority($priority)
-    {
-        $this->getSwiftMessage()->setPriority($priority);
-
-        return $this;
-    }
-
-    /**
-     * @param string $path Path to file
-     * @param string $alias An override for the name of the file
-     * @param string $mime The mime type for the attachment
-     * @return $this
+     * Get template data
+     *
+     * Hardcoded data - these used to be public methods in v4 when Email extended ViewableData:
+     * IsEmail: used in template detect if rendering an email template rather than a page template
+     * BaseUrl: used in template to get the base URL for the email
      */
-    public function addAttachment($path, $alias = null, $mime = null)
+    public function getData(): array|ViewableData
     {
-        $attachment = \Swift_Attachment::fromPath($path);
-        if ($alias) {
-            $attachment->setFilename($alias);
-        }
-        if ($mime) {
-            $attachment->setContentType($mime);
+        $extraData = [
+            'IsEmail' => true,
+            'BaseURL' => Director::absoluteBaseURL(),
+        ];
+        if (is_array($this->data)) {
+            return array_merge($this->data, $extraData);
         }
-        $this->getSwiftMessage()->attach($attachment);
-
-        return $this;
-    }
-
-    /**
-     * @param string $data
-     * @param string $name
-     * @param string $mime
-     * @return $this
-     */
-    public function addAttachmentFromData($data, $name, $mime = null)
-    {
-        $attachment = new \Swift_Attachment($data, $name);
-        if ($mime) {
-            $attachment->setContentType($mime);
+        $data = clone $this->data;
+        foreach ($extraData as $key => $value) {
+            $data->{$key} = $value;
         }
-        $this->getSwiftMessage()->attach($attachment);
-
-        return $this;
+        return $data;
     }
 
     /**
-     * @return array|ViewableData The template data
+     * Set template data
      */
-    public function getData()
-    {
-        return $this->data;
-    }
-
-    /**
-     * @param array|ViewableData $data The template data to set
-     * @return $this
-     */
-    public function setData($data)
+    public function setData(array|ViewableData $data)
     {
         $this->data = $data;
-        $this->invalidateBody();
-
         return $this;
     }
 
     /**
-     * @param string|array $name The data name to add or array to names => value
-     * @param string|null $value The value of the data to add
-     * @return $this
+     * Add a a single or multiple pieces of template data
+     * $name can be either the name to add, or an array of [name => value]
```

Comment:
```suggestion
     * Add data to be used in the template
     * @param string|array $nameOrData can be either the name to add, or an array of [name => value]
```
"A single or multiple" is a complicated way of describing this.
Also describing parameter usage should be done using `@param`

Reply:
Updated